src/include/duckdb.h

```diff
@@ -2322,6 +2322,30 @@ Must be destroyed with `duckdb_destroy_value`.
 */
 DUCKDB_API duckdb_value duckdb_create_array_value(duckdb_logical_type type, duckdb_value *values, idx_t value_count);
 
+/*!
+Creates a map value from a map type and two arrays of keys and values of length `count`.
+Must be destroyed with `duckdb_destroy_value`.
+
+* @param map_type The type of the map
+* @param keys The keys for the map
+* @param values The values for the map
+* @param count The number of key-value pairs in the map
+* @return The map value, or nullptr, if the child type is `DUCKDB_TYPE_ANY` or `DUCKDB_TYPE_INVALID`.
+*/
+DUCKDB_API duckdb_value duckdb_create_map_value(duckdb_logical_type map_type, duckdb_value *keys, duckdb_value *values,
```

Comment:
Maybe this could be two list values instead? (i.e. just take a `duckdb_value keys, duckdb_value values` where the values are lists?)

Reply:
@Mytherin I changed the function to take `duckdb_value` instead of `duckdb_value *`. Also resolved merge conflicts.

Reply:
@Mytherin I have another PR (https://github.com/duckdb/duckdb/pull/14602) too (helper functions for map and union). Can you review that when you get a chance? I'll be happy to make any changes that are needed. Thank you.